web/src/message_edit.js

```diff
@@ -1280,13 +1283,35 @@ function handle_message_move_failure_due_to_time_limit(xhr, handle_confirm, on_h
     });
 }
 
+function show_message_moved_toast(toast_params) {
+    const new_stream_name = sub_store.maybe_get_stream_name(toast_params.new_stream_id);
+    const stream_topic = `#**${new_stream_name}>${toast_params.new_topic_name}**`;
+    feedback_widget.show({
+        populate($container) {
+            const rendered_msg = $t_html(
+                {
+                    defaultMessage: "Message moved to <z-link>{stream_topic}</z-link>",
+                },
+                {
+                    stream_topic,
+                    "z-link": (content) =>
+                        `<a href="${hash_util.by_stream_topic_url(toast_params.new_stream_id, toast_params.new_topic_name)}">${content}</a>`,
+                },
```

Comment:
I'm worried about this approach for constructing the link introducing an XSS issue. with content in that `href`; do we use this pattern elsewhere? Generally best practice would be to do this via a handlebars template, where one can easily separate the part that is a URL (should be escaped) from the `a` tag and other HTML elements (should be processed as HTML).

Reply:
Understood, I've shifted it to a handlebars template.